connectors/sources/mysql.py

```diff
@@ -111,31 +117,74 @@ async def ping(self):
+            list: Column names and query response
+        """
 
-                while True:
-                    rows = await cursor.fetchmany(size=size)
+        size = int(self.configuration.get("fetch_size", DEFAULT_FETCH_SIZE))
+        logger.debug(f"Streaming {database}.{table} {size} rows at a time")
 
-                    if len(rows) == 0:
+        # retry: Current retry counter
+        # yield_once: Yield(fields) once flag
+        retry = yield_once = 1
+
+        # rows_fetched: Rows fetched counter
+        # cursor_position: Mocked cursor position
+        rows_fetched = cursor_position = 0
+        while retry <= int(self.configuration.get("retry_count", DEFAULT_RETRY_COUNT)):
+            try:
+                async with self.connection_pool.acquire() as connection:
+                    async with connection.cursor(aiomysql.cursors.SSCursor) as cursor:
+                        await cursor.execute(query)
+
+                        # sending back column names only once
+                        if yield_once:
+                            yield [column[0] for column in cursor.description]
+                            yield_once = 0
+
+                        # setting cursor position where it was failed
+                        if cursor_position:
+                            await cursor.scroll(cursor_position, mode="absolute")
+                            logger.debug(
+                                f"Cursor moved to {cursor_position} position for {database}.{table}"
+                            )
+
+                        while True:
+                            rows = await cursor.fetchmany(size=size)
+                            rows_length = len(rows)
+
+                            # resetting cursor position & retry to 0 for next batch
+                            if cursor_position:
+                                cursor_position = retry = 0
+
+                            if not rows_length:
+                                break
+
+                            for row in rows:
+                                yield row
+
+                            rows_fetched += rows_length
+                            await asyncio.sleep(0)
                         break
-
-                    for row in rows:
-                        yield row
-
-                    count += len(rows)
-                    if count > 100:
-                        logger.info(f"Collected {count} rows in {database}.{table}")
-                        count = 0
-                    await asyncio.sleep(0)
+            except IndexError as exception:
+                logger.exception(
+                    f"None of rows fetched from {rows_fetched} rows in {database}.{table}. Exception: {exception}"
+                )
+                break
+            except Exception as exception:
+                logger.exception(
+                    f"Retry count: {retry} out of {DEFAULT_RETRY_COUNT} for rows in {database}.{table}. Exception: {exception}"
```

Comment:
if `retry_count` is differerent from `DEFAULT_RETRY_COUNT` this log info won't show the current countdown